src/plugins/data/server/search/session/session_service.ts

```diff
@@ -38,24 +50,129 @@ export type ISearchSessionClient = ReturnType<
   ReturnType<BackgroundSessionService['asScopedProvider']>
 >;
 
+export interface SessionInfo {
+  insertTime: Moment;
+  retryCount: number;
+  ids: Map<string, string>;
+}
+
 export class BackgroundSessionService {
   /**
    * Map of sessionId to { [requestHash]: searchId }
    * @private
    */
-  private sessionSearchMap = new Map<string, Map<string, string>>();
+  private sessionSearchMap = new Map<string, SessionInfo>();
+  private internalSavedObjectsClient!: SavedObjectsClientContract;
+  private monitorInterval!: NodeJS.Timeout;
+
+  constructor(private readonly logger: Logger) {}
 
-  constructor() {}
+  private setupMonitoring = (savedObjects: SavedObjectsServiceStart) => {
+    // TODO: setup monitoring only if BGS is enabled
+    const internalRepo = savedObjects.createInternalRepository();
```

Comment:
The `BACKGROUND_SESSION_TYPE` type is marked as `hidden`, which means that the internal repository can't access these saved objects by default. I'd expect your bulkGet/bulkUpdate calls below to return an [unsupported type error](https://github.com/elastic/kibana/blob/d4b2a5145a9aa813146e2c587f7be2929f09fb7e/src%2Fcore%2Fserver%2Fsaved_objects%2Fservice%2Flib%2Ferrors.ts#L96-L101) as part of its response when this happens. Are you seeing different behavior?